code/game/gamemodes/traitor/double_agents.dm

```diff
@@ -238,6 +246,11 @@
 		state.add_steal_targets_timer()
 		if(!issilicon(traitor.current))
 			give_pinpointer(traitor)
+		//Optional traitor objective
+		if(prob(PROB_ACTUAL_TRAITOR))
+			traitor.special_role = TRAITOR_AGENT_ROLE
+			traitor.licence_to_kill = TRUE

```

Comment:
That's how I did it originally, but I thought this way would be better to allow for an easier transition to a non role-based framework (IAAchan when). Nonetheless, I'll revert it